.github/workflows/master_build.yml

```diff
@@ -0,0 +1,114 @@
+name: Master CI Build
+
+on:
+  push:
+    branches:
+#      - master
+      - translation-experiment
+
+jobs:
+  build_en_docs:
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        python-version: [3.7]
+
+    name: Build and deploy english docs
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -U Sphinx==2.4.4
+        pip install --upgrade recommonmark
+        pip install sphinx_rtd_theme
+    - name: Build documentation
+      shell: bash
+      run: |
+        cd docs
+        make html
+        mkdir -p build-archives
+        cd build/html/
+        tar -czvf ../../build-archives/build-en.tar.gz *
+    - name: Upload documentation
+      uses: JulianGro/SFTP-Deploy-Action@v1.4
+      with:
+        username: 'gha'
+        server: '94.130.177.235'
+        ssh_private_key: ${{ secrets.SSH_PRIVATE_KEY }}
+        local_path: 'docs/build-archives/*'
+        remote_path: '/var/www/html/gha/'
+        args: '-o ConnectTimeout=5'
+    - name: Deploy english documentation
+      shell: bash
+      run: |
+        printf "%s" "${{ secrets.SSH_PRIVATE_KEY }}" >private_key.pem
+        chmod 600 private_key.pem
+        ssh -o StrictHostKeyChecking=no -i private_key.pem gha@94.130.177.235 mkdir -p /var/www/html/gha/
+        ssh -o StrictHostKeyChecking=no -i private_key.pem gha@94.130.177.235 tar -xvzf /var/www/html/gha/build-en.tar.gz -C /var/www/html/gha/
+        ssh -o StrictHostKeyChecking=no -i private_key.pem gha@94.130.177.235 rm /var/www/html/gha/build-en.tar.gz
```

Comment:
You're connecting 3 times to the ssh here. How about running the ssh command once while sending multiple commands perhaps using here docs https://bash.cyberciti.biz/guide/Here_documents or more ssh specific help page https://www.cyberciti.biz/faq/linux-unix-osx-bsd-ssh-run-command-on-remote-machine-server/ , which would make it look like
```
ssh -o StrictHostKeyChecking=no -i private_key.pem gha@94.130.177.235 << EOF
    mkdir -p /var/www/html/gha/
    tar -xvzf /var/www/html/gha/build-en.tar.gz -C /var/www/html/gha/
    rm /var/www/html/gha/build-en.tar.gz
EOF
```

Reply:
As I don't think those commands are run as a bash script, but rather just as bash commands, I changed them to one-liners.